tokio-util/src/codec/decoder.rs

```diff
@@ -114,14 +128,20 @@ pub trait Decoder {
     ///
     /// This method defaults to calling `decode` and returns an error if
     /// `Ok(None)` is returned while there is unconsumed data in `buf`.
-    /// Typically this doesn't need to be implemented unless the framing
-    /// protocol differs near the end of the stream.
+    /// Typically this doesn't need to be implemented unless you want to
+    /// create frames from sources that can be resumed after an eof,
+    /// such as files of FIFOs, on in cases where the framing protocol
+    /// differs near the end of the stream.
```

Comment:
Do you _have_ to implement `decode_eof` to support streams that can be resumed? I assume not.

Reply:
Yeah this should be more clear! It only needs to be implemented if you want to read ACROSS resumes, as the default implementation throws an error when the buffer is not empty after an eof.
I'm gonna rewrite it to be more clear and also fix that typo 😄